tests/foreman/cli/test_classparameters.py

```diff
@@ -16,107 +16,253 @@
 @Upstream: No
 """
 
-from robottelo.decorators import run_only_on, stubbed, tier1, tier2, tier3
+import json
+from random import choice
+from robottelo import ssh
+from robottelo.cli.base import CLIReturnCodeError
+from robottelo.cli.environment import Environment
+from robottelo.cli.factory import make_hostgroup
+from robottelo.cli.host import Host
+from robottelo.cli.hostgroup import HostGroup
+from robottelo.cli.proxy import Proxy
+from robottelo.cli.puppet import Puppet
+from robottelo.cli.smartclass import SmartClassParameter
+from robottelo.config import settings
+from robottelo.datafactory import filtered_datapoint, gen_string
+from robottelo.decorators import (
+    run_in_one_thread,
+    run_only_on,
+    skip_if_bug_open,
+    stubbed,
+    tier1,
+    tier2,
+)
 from robottelo.test import CLITestCase
 
 
+@filtered_datapoint
+def valid_sc_parameters():
+    """Returns a list of valid smart class parameter types and values"""
+    return [
+        {
+            u'sc_type': 'string',
+            u'value': gen_string('utf8'),
+        },
+        {
+            u'sc_type': 'boolean',
+            u'value': choice(['0', '1']),
+        },
+        {
+            u'sc_type': 'integer',
+            u'value': gen_string('numeric', 5),
+        },
+        {
+            u'sc_type': 'real',
+            u'value': -123.0,
+        },
+        {
+            u'sc_type': 'array',
+            u'value': "['ZvqqjmtkeW', 'SGnfs123yekPk']",
+        },
+        {
+            u'sc_type': 'hash',
+            u'value': {'a': 'test'},
+        },
+        {
+            u'sc_type': 'yaml',
+            u'value': 'name=>XYZ',
```

Comment:
better to define a constant/permanent file having yaml data or somewhere in constants.py.